nilearn/_utils/niimg.py

```diff
@@ -0,0 +1,144 @@
+"""
+Neuroimaging file input and output.
+"""
+# Author: Gael Varoquaux, Alexandre Abraham, Philippe Gervais
+# License: simplified BSD
+
+import copy
+import gc
+import collections
+from six import string_types
+
+import numpy as np
+import nibabel
+
+from .numpy_conversions import as_ndarray
+
+
+def _safe_get_data(img):
+    """ Get the data in the image without having a side effect on the
+        Nifti1Image object
+    """
+    if hasattr(img, '_data_cache') and img._data_cache is None:
+        # By loading directly dataobj, we prevent caching if the data is
+        # memmaped. Preventing this side-effect can save memory in some cases.
+        img = copy.deepcopy(img)
+    # typically the line below can double memory usage
+    # that's why we invoke a forced call to the garbage collector
+    gc.collect()
+    return img.get_data()
+
+
+def load_img(img, dtype=None):
+    """Load a niimg, check if it is a nibabel SpatialImage and cast if needed
+
+    Parameters:
+    -----------
+
+    img: Niimg-like object
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        Image to load.
+
+    Returns:
+    --------
+    img: image
+        A loaded image object.
+    """
+    if isinstance(img, string_types):
+        # data is a filename, we load it
+        img = nibabel.load(img)
+    elif not isinstance(img, nibabel.spatialimages.SpatialImage):
+        raise TypeError("Data given cannot be loaded because it is"
+                        " not compatible with nibabel format:\n"
+                        + short_repr(img))
+    return img
+
+
+def new_img_like(ref_img, data, affine, copy_header=False):
+    """Create a new image of the same class as the reference image
+
+    Parameters
+    ----------
+    ref_img: image
+        Reference image. The new image will be of the same type.
+
+    data: numpy array
+        Data to be stored in the image
+
+    affine: 4x4 numpy array
+        Transformation matrix
+
+    copy_header: boolean, optional
+        Indicated if the header of the reference image should be used to
+        create the new image
+
+    Returns
+    -------
+
+    new_img: image
+        An image which has the same type as the reference image.
+    """
+    data = np.asarray(data)
+    if data.dtype == bool:
+        data = as_ndarray(data, dtype=np.int8)
+    header = None
+    if copy_header:
+        header = copy.copy(ref_img.get_header())
+        header['scl_slope'] = 0.
+        header['scl_inter'] = 0.
+        header['glmax'] = 0.
+        header['cal_max'] = np.max(data) if data.size > 0 else 0.
+        header['cal_max'] = np.min(data) if data.size > 0 else 0.
```

Comment:
Could you use the nibabel update header function here (with a try/except if it's not available in old nibabel versions).